environment/service.go

```diff
@@ -0,0 +1,141 @@
+package environment
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/dgrijalva/jwt-go"
+	"github.com/fabric8-services/fabric8-common/log"
+	"github.com/fabric8-services/fabric8-tenant/environment/generated"
+	"github.com/fabric8-services/fabric8-tenant/toggles"
+	"github.com/fabric8-services/fabric8-tenant/utils"
+	goajwt "github.com/goadesign/goa/middleware/security/jwt"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+)
+
+//go:generate go-bindata -prefix "./templates/" -pkg templates -o ./generated/templates.go ./templates/...
+
+const (
+	f8TenantServiceRepoUrl = "https://github.com/fabric8-services/fabric8-tenant"
+	rawFileURLTemplate     = "%s/blob/%s/%s"
+	templatesDirectory     = "environment/templates/"
+)
+
+var DefaultEnvTypes = []string{"che", "jenkins", "user", "run", "stage"}
+
+var templateNames = map[string][]*Template{
+	"run":     tmpls(runParams, "fabric8-tenant-deploy.yml"),
+	"stage":   tmpls(stageParams, "fabric8-tenant-deploy.yml"),
+	"che-mt":  tmpls(noParams, "fabric8-tenant-che-mt.yml", "fabric8-tenant-che-quotas.yml"),
+	"che":     tmpls(noParams, "fabric8-tenant-che.yml", "fabric8-tenant-che-quotas.yml"),
+	"jenkins": tmpls(noParams, "fabric8-tenant-jenkins.yml", "fabric8-tenant-jenkins-quotas.yml"),
+	"user":    tmpls(noParams, "fabric8-tenant-user.yml"),
+}
+
+func tmpls(defaultParams map[string]string, filenames ...string) []*Template {
+	tmpls := make([]*Template, 0, len(filenames))
+	for _, fileName := range filenames {
+		tmpls = append(tmpls, newTemplate(fileName, defaultParams))
+	}
+	return tmpls
+}
+
+type Service struct {
+	ctx               context.Context
+	templatesRepo     string
+	templatesRepoBlob string
+	templatesRepoDir  string
+}
```

Comment:
`ctx` as field, make sense ?? OR ctx should be passed as arg to methods for this service.

Reply:
good point :+1: